go/vt/mysqlctl/backup.go

```diff
@@ -315,21 +315,28 @@ func removeExistingFiles(cnf *Mycnf) error {
 
 // ShouldRestore checks whether a database with tables already exists
 // and returns whether a restore action should be performed
-func ShouldRestore(ctx context.Context, params RestoreParams) (bool, error) {
-	if params.DeleteBeforeRestore || RestoreWasInterrupted(params.Cnf) {
+func ShouldRestore(
```

Comment:
previously, this function was called by `tabletmanager.restoreDataLocked` after constructing `mysqlctl.RestoreParams`. That is (a) unnecessary and (b) makes it difficult to share this function with `tabletmanager.CloneTablet`.